example/generator/odp_generator.c

```diff
@@ -1210,10 +1214,35 @@ int main(int argc, char *argv[])
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
+	if (odp_pool_capability(&pool_capa)) {
+		ODPH_ERR("Error: pool capability failed\n");
+		return -1;
+	}
+
+	pkt_len = (args->appl.payload + MAX_HDR_LEN) > POOL_PKT_LEN ?
+			(args->appl.payload + MAX_HDR_LEN) : POOL_PKT_LEN;
+
+	if (pool_capa.pkt.max_len && pkt_len > pool_capa.pkt.max_len)
+		pkt_len = pool_capa.pkt.max_len;
+
+	/* Check weather we have sufficient segments to support requested packet
+	 * length, if not adjust to bigger segment size */
+	if ((seg_len * pool_capa.pkt.max_segs_per_pkt) < pkt_len)
+		seg_len = pkt_len / pool_capa.pkt.max_segs_per_pkt;
```

Comment:
This is unnecessary and should not be needed by any ODP implementation that complies with the API specification. In the pool creation parameters the seg_len does not specify the exact segment length that must be used but "Minimum number of packet data bytes that can be stored in the first segment of a newly allocated packet". IOW, an implementation should not get confused and not able to create a packet pool if an application does not require may contiguous bytes in the beginning of packet data.

Reply:
Will remove this code